### PR TITLE
Some basic docstrings for Cohort() and EventFrame().

### PR DESCRIPTION
**Pull request by CarolineMorton**
Some basic docstring for Cohort() and EventFrame().

### CODE REVIEW

File: cohortextractor2/dsl.py [hunk 1]
```diff
@@ -59,10 +59,17 @@
...
+    Represents the cohort of patients in the defined study.
+    """
 
     def set_population(self, population: PatientSeries) -> None:
-        """Set the population variable for this cohort."""
+        """
+        This sets the population that are included within the Cohort
+
+        Args:
+            population: Patients who are included within the Cohort
```
Comment (benbc) [inline cohortextractor2/dsl.py:71]: This isn't strictly a series containing the patients. It's a boolean series and whether the boolean is true or false dictates whether that patient is included.

**Review by benbc — APPROVED** [review]
This looks good. Just one note on a detail -- you could definitely merge it as-is to get it in before Steve's PR.

There are a few things that you've documented that aren't part of the DSL surface that study authors will use (for example `raise_category_errors`). Of course it's fine to document them, but don't worry to much about making them perfect (later we'll reorganize the code so that there's no chance of study authors stumbling across things like this).

File: cohortextractor2/dsl.py [hunk 2]
```diff
@@ -115,18 +150,41 @@ def filter(  # noqa: A003
...
+        Returns:
+            SortedEventFrame with a given sort column
+        """
         return SortedEventFrame(self.qm_table, *columns)
 
     def count_for_patient(self) -> PatientSeries:
-        """Return a PatientSeries with count_for_patient of matching events per patient."""
+        """
+        Take the information from the multiple row per patient EventFrame and counts
+        the events per patient.
```
Comment (madwort) [inline cohortextractor2/dsl.py:168]: typo - either `take` or `counts`

File: cohortextractor2/dsl.py [hunk 3]
```diff
@@ -115,18 +150,41 @@ def filter(  # noqa: A003
...
+        Returns:
+            PatientSeries: A count of events per patient
+        """
         return PatientSeries(self.qm_table.count())
 
     def exists_for_patient(self) -> PatientSeries:
-        """Return a PatientSeries indicating whether each patient has a matching event."""
+        """
+        Take the information from the multiple row per patient EventFrame and counts
+        the events per patient.
```
Comment (madwort) [inline cohortextractor2/dsl.py:181]: seems like an incorrect definition?

File: cohortextractor2/dsl.py [hunk 4]
```diff
@@ -227,6 +285,15 @@ class Column:
 
 
 def not_null_patient_series(patient_series: PatientSeries) -> PatientSeries:
+    """
+    Removes all null values from a PatientSeries.
```
Comment (madwort) [inline cohortextractor2/dsl.py:289]: This implies the original PatientSeries is modified in-place, contradicting `Returns`

File: cohortextractor2/dsl.py [hunk 5]
```diff
@@ -227,6 +285,15 @@ class Column:
...
+
+    Args:
+        patient_series: PatientSeries being checked for null values
```
Comment (madwort) [inline cohortextractor2/dsl.py:292]: how about `PatientSeries that may contain null values`?

File: cohortextractor2/dsl.py [hunk 6]
```diff
@@ -73,7 +80,13 @@ def set_population(self, population: PatientSeries) -> None:
             )
 
     def add_variable(self, name: str, variable: PatientSeries) -> None:
-        """Add a variable to this cohort by name."""
+        """
+        Add a variable to this Cohort with a given name.
+
+        Args:
+            name: The name of the variable being added
+            variable: The PatientSeries being added as the named variable.
```
Comment (madwort) [inline cohortextractor2/dsl.py:88]: personal preference: I like `to add` here instead of `being added`, but ymmv

File: cohortextractor2/dsl.py [hunk 7]
```diff
@@ -59,10 +59,17 @@
...
 class Cohort:
-    """Represents the cohort of patients in a study."""
+    """
+    Represents the cohort of patients in the defined study.
+    """
 
     def set_population(self, population: PatientSeries) -> None:
-        """Set the population variable for this cohort."""
+        """
+        This sets the population that are included within the Cohort
```
Comment (madwort) [inline cohortextractor2/dsl.py:68]: nit: I would rewrite to remove `this` - how about `Set the population to be included...`?

**Review by madwort — COMMENTED** [review]
typos and nitpicks for style

File: cohortextractor2/dsl.py (hunk 3 above)
Comment (CarolineMorton) [reply cohortextractor2/dsl.py:181]: Thanks - I have corrected.

File: cohortextractor2/dsl.py (hunk 4 above)
Comment (CarolineMorton) [reply cohortextractor2/dsl.py:289]: thaks - hopefully cleared this up

**Pull request merged**